### PR TITLE
March 2021 updates to

### PR DESCRIPTION
**Pull request by woodjt**
PBP dashboarding project, git ignore file (.vscode), gcs_utils (prms_setup), the dataflow_utils (change_data_types) file, and env.example (put everything in alphabetical order)

### CODE REVIEW

File: dags/police_demographics_airflow.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    task_id='police_demographics_dataflow',
+    bash_command="python {}/dependencies/dataflow_scripts/police_demographics_dataflow.py --input gs://{}_police/"
+                 "officer_demographics/".format(os.environ['DAGS_PATH'], os.environ['GCS_PREFIX']) + "{{ ds|get_ds_year }}/"
+                 "{{ ds|get_ds_month }}/{{ ds }}_demographics.json --avro_output " + "gs://{}_police/"
+                 "officer_demographics/avro_output/".format(os.environ['GCS_PREFIX']) + "{{ ds|get_ds_year }}/"
+                 "{{ ds|get_ds_month }}/{{ ds }}/",
+    dag=dag
+)
+
+# this table gets overrwritten since it will contain newly hired officers and ALL previously employed officers. It would be possible to first search officers that are already in the database, and only add officers that are not present.  However, this is unnecessarily complicated at the present.
```
Comment (jtotoole) [inline dags/police_demographics_airflow.py:36]: smallest of nitpicks: for comments like this that have notes particular to a moment in time, it's useful to add a date

File: env.example [hunk 2]
```diff
@@ -1,51 +1,71 @@
...
+FIRE_MSSQL_UN=
+GCLOUD_PROJECT=
+GCP_CONN_ID=
+GCS_LOADERS_PATH=
+GCS_PREFIX=
+GITHUB_TOKEN=
 GKE_CLUSTER=
-ISAT_UN=
-ISAT_PW=
+GOOGLE_APPLICATION_CREDENTIALS=
```
Comment (jtotoole) [inline env.example:34]: unrelated to this PR, but i'm realizing now that we can remove this line

File: env.example [hunk 3]
```diff
@@ -1,51 +1,71 @@
...
+COUNTY_PW=
 COUNTY_REFRESH_TOKEN=
-COUNTY_BASE_URL=https://moveit.alleghenycounty.us/api/v1/swagger
-DAGS_PATH=/your/system/path/data-rivers/dags
-DWLIVE_UN=
-DWLIVE_PW=
+COUNTY_UN=
+CREDENTIALS=
+DAGS_PATH=
+DATAFLOW_PATH=
```
Comment (jtotoole) [inline env.example:18]: this isn't necessary given that we've already got [`SETUP_PY_DATAFLOW`](https://github.com/CityofPittsburgh/data-rivers/blob/master/env.example#L45)

File: env.example [hunk 4]
```diff
@@ -1,51 +1,71 @@
...
 COUNTY_ACCESS_TOKEN=
+COUNTY_BASE_URL=
+COUNTY_PW=
 COUNTY_REFRESH_TOKEN=
-COUNTY_BASE_URL=https://moveit.alleghenycounty.us/api/v1/swagger
-DAGS_PATH=/your/system/path/data-rivers/dags
-DWLIVE_UN=
-DWLIVE_PW=
+COUNTY_UN=
+CREDENTIALS=
```
Comment (jtotoole) [inline env.example:16]: not necessary

File: env.example [hunk 5]
```diff
@@ -1,51 +1,71 @@
 ACCELA_CLIENT_ID=
 ACCELA_CLIENT_SECRET=
-ACCELA_USER=
 ACCELA_PW=
+ACCELA_USER=
 AIRFLOW_WEB_SERVER=
```
Comment (jtotoole) [inline env.example:5]: not neccessary

File: env.example [hunk 6]
```diff
@@ -1,51 +1,71 @@
...
-GCS_PREFIX=pghpa_test
+FIRE_MSSQL_HOST=
+FIRE_MSSQL_PW=
+FIRE_MSSQL_UN=
+GCLOUD_PROJECT=
+GCP_CONN_ID=
+GCS_LOADERS_PATH=
+GCS_PREFIX=
+GITHUB_TOKEN=
 GKE_CLUSTER=
```
Comment (jtotoole) [inline env.example:33]: not necessary

File: env.example [hunk 7]
```diff
@@ -1,51 +1,71 @@
...
 FIRE_MSSQL_DB=
-GCP_CONN_ID=gs://pghpa_service_acct_creds/data_rivers_key.json
-GCLOUD_PROJECT=data-rivers-testing
-GCS_PREFIX=pghpa_test
+FIRE_MSSQL_HOST=
+FIRE_MSSQL_PW=
+FIRE_MSSQL_UN=
+GCLOUD_PROJECT=
+GCP_CONN_ID=
+GCS_LOADERS_PATH=
```
Comment (jtotoole) [inline env.example:30]: not necessary

File: env.example [hunk 8]
```diff
@@ -1,51 +1,71 @@
...
+GOOGLE_APPLICATION_CREDENTIALS=
 ISAT_DB=
+ISAT_HOST=
+ISAT_PW=
+ISAT_UN=
 MONDAY_API_TOKEN=
 MSTEAMS_WEBHOOK=
-OTRS_USER=
-OTRS_PW=
+OJDBC_JAR_PATH=
```
Comment (jtotoole) [inline env.example:41]: If you want to abstract this, I'd refactor elsewhere to use `os.environ['OJDBC_JAR_PATH']` in all the `.py` files [here](https://github.com/CityofPittsburgh/data-rivers/search?q=jar), then change it in the Cloud Composer interface. Personally I could go either way on it.

**Comment by jtotoole** [comment]
Looking good! I added some notes related to unnecessary variables in `env.example` (most of which aren't related to this PR), but the substance of this work is 🎯

File: env.example (hunk 8 above)
Comment (woodjt) [reply env.example:41]: That makes sense. I'll hold back on adding more refactors to the to-do list for now. I will revisit this in the future. Thanks!

File: env.example (hunk 5 above)
Comment (jtotoole) [reply env.example:4]: oops sorry that's my bad; the one we don't need is `AIRFLOW_WEB_SERVER`; we do want [`ACCELA_USER`](https://github.com/CityofPittsburgh/data-rivers/blob/master/dags/dependencies/gcs_loaders/accela_gcs.py#L26)

**Pull request merged**